src/Worker.php

```diff
@@ -81,81 +71,82 @@ protected function tick($queue)
             return true;
         }
 
-        $message = $this->serializer->deserialize($packet);
         try {
-            if ($this->invoke($message) === false) {
-                $this->jobShutdown($message);
+            $options = unserialize($packet);
+
+            if ($this->invoke($options) === false) {
+                $this->jobShutdown($options);
                 return false;
             }
         } catch (Exception $exception) {
-            $this->jobException($message, $exception);
+            $this->jobException($options, $exception);
         }
 
         return true;
     }
 
     /**
-     * Invoke the messages handler
+     * Invoke the options command
      *
-     * @param Message $message
+     * @param AbstractOptions $options
      *
      * @return null|bool
      */
-    private function invoke(Message $message)
+    private function invoke(AbstractOptions $options)
     {
-        $this->jobStart($message);
+        $this->jobStart($options);
 
-        $result = call_user_func(
-            $this->handlers->get($message->handler()),
-            $message
-        );
+        $result = $this->commands
+            ->make($options->command())
+            ->withOptions($options)
```

Comment:
Maybe `copyWithOptions()` should be added to `CommandInterface` as a public method ? At that point, assuming it was changed to be `public` in `CommandImmutableOptionsTrait`, you could simply `use CommandImmutableOptionsTrait` to implement it.